provider/acme/provider.go

```
@@ -130,8 +131,19 @@ func (p *Provider) Init(_ types.Constraints) error {
 	}
 
 	// Reset Account if caServer changed, thus registration URI can be updated
-	if p.account != nil && p.account.Registration != nil && !strings.HasPrefix(p.account.Registration.URI, p.CAServer) {
-		p.account = nil
+	if p.account != nil && p.account.Registration != nil {
+		aru, err := url.Parse(p.account.Registration.URI)
+		if err != nil {
+			log.Error("Unable to parse account.Registration URL")
+		}
+		cau, err := url.Parse(p.CAServer)
+		if err != nil {
+			log.Error("Unable to parse CAServer URL")
```

You need to return an error